### PR TITLE
Avoid allocation in build_scriptint

### PR DESCRIPTION
**Pull request by stevenroose**
Hehe, reason for party, let's invite apoelstra !

### CODE REVIEW

**Comment by apoelstra** [comment]
I like this approach -- but I think there's a more elegant way to do this, where `write_scriptint` takes an arbitrary `W: Write`, doesn't do its own length-tracking, and the caller gives it a `Cursor<&mut [u8]>` to get the length-tracking.

**Review by apoelstra — APPROVED** [review]
ACK c80dbc21692947278ce1c08d9b6b85f05620ceef

acking this version, as not to let the perfect be the enemy of the good. It's great to eliminate an allocation!

File: src/blockdata/script.rs [hunk 1]
```diff
@@ -1106,7 +1114,7 @@ mod test {
     use core::str::FromStr;
 
     use super::*;
-    use super::build_scriptint;
+    use super::write_scriptint;
```
Comment (tcharding) [inline src/blockdata/script.rs:1117]: We can just delete these lines, wildcard import right above :) Probably not worth requiring re-acks just to do this though.

**Review by tcharding — APPROVED** [review]
ACK c80dbc21692947278ce1c08d9b6b85f05620ceef

**Review by sanket1729 — APPROVED** [review]
Code review ACK

**Pull request merged**